src/codegen.cpp

```diff
@@ -709,10 +709,11 @@ typedef struct {
 
 typedef struct {
     size_t len;
+    // variable length
     struct {
         int64_t isref;
         Function *f;
-    } data[];
+    } data[1];
```

Comment:
perhaps we should just switch this struct to a `std::unordered_map< uint64_t, Function* >`